File: transfer-backend/src/main/java/br/com/transfer/service/TransferenciaService.java
package br.com.transfer.service;

import java.util.List;

import br.com.transfer.entity.Transferencia;

public interface TransferenciaService {
	
	Transferencia createTransferencia(Integer contaBancariaOrigemId, Integer contaBancariaDestinoId, Transferencia transferencia);

    Transferencia updateTransferencia(Integer transferenciaId, Transferencia transferencia);

    void deleteTransferencia(Integer transferenciaId);

    List<Transferencia> getTransferencias();

    Transferencia getTransferenciaById(Integer transferenciaId);

}

File: transfer-frontend/src/services/TransferenciaService.js
import axios from "axios";

const TRANSFERENCIAS_ENDPOINT = "http://localhost:8080/api/transferencias/"

class TransferenciaService {
    createTransferencia(contaBancariaOrigemId, contaBancariaDestinoId, transferencia){
        return axios.post(TRANSFERENCIAS_ENDPOINT + contaBancariaOrigemId + "/" + contaBancariaDestinoId, transferencia);
    }
    updateTransferencia(transferenciaId, transferencia){
        return axios.put(TRANSFERENCIAS_ENDPOINT + transferenciaId, transferencia);
    }
    deleteTransferencia(transferenciaId){
        return axios.delete(TRANSFERENCIAS_ENDPOINT + transferenciaId);
    }
    getTransferencias() {
        return axios.get(TRANSFERENCIAS_ENDPOINT);
    }
    getTransferenciaById(transferenciaId){
        return axios.get(TRANSFERENCIAS_ENDPOINT + transferenciaId);
    }   
}

export default new TransferenciaService()
